File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    
    bool areBad(char a, char b)
    {
        if (abs(a - b) == 32)
            return true;
        return false;
    }
    
    string makeGood(string s) {
        
        stack<char> stack;
        
        int i = 0;
        
        while(i < s.size())
        {
            while(i < s.size() && stack.size() > 0 and areBad(s[i], stack.top()))
            {
                stack.pop();
                i++;
            }
            
            if(i < s.size())
                stack.push(s[i]);
            i++;
            
        }
        
        cout<<'a' - 'A';
        
        string ans = "";
        
        while(!stack.empty())
        {
            ans += stack.top();
            stack.pop();
        }
        
        reverse(ans.begin(), ans.end());
        return ans;
        
    }
};

File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        
        stack = deque()
        
        i = 0
        
        
        def isBad(a, b):
            return abs(ord(a) - ord(b)) == 32
        
        
        while i < len(s):
            
            while i < len(s) and len(stack) > 0 and isBad(s[i], stack[-1]):
                stack.pop()
                i+=1
            
            if i < len(s):
                stack.append(s[i])
                i += 1
                
        
        return "".join(list(stack))